### ISSUE TITLE
Color maps can look wrong

### ISSUE THREAD

**Issue by NeatNit** [ChristopherChudzicki/math3d-react#228]
https://www.math3d.org/wiLC2pTt

What's happening right now: for each sample point, it calculates the color of that point. When creating the surface, it interpolates _the color_ of the points in order to color the surface.

Problems: this can cause strange patterns like the weird diagonal dark shades in the spinning flat surface in the example above:

![image](https://user-images.githubusercontent.com/2400026/68924270-0e4c0b00-0789-11ea-880f-4917f2d39070.png)

It can also result in wrong colors, especially with low sample rates. The two `abs(y)` surfaces are the same, but one of them has 65 samples and other other has 5 samples. I expect them to look the same, but the low-samples one looks wrong:

![image](https://user-images.githubusercontent.com/2400026/68924204-e9f02e80-0788-11ea-9e68-a5df652510ce.png)

Possible solution: interpolate _the value_ between points instead of the color, and apply the color map on that.

**Comment by ChristopherChudzicki** [comment]
> What's happening right now: for each sample point, it calculates the color of that point. When creating the surface, it interpolates the color of the points in order to color the surface.

Yes, that is entirely correct. The possible solution you suggest:

> interpolate the value between points instead of the color, and apply the color map on that.

Is already achieved by just increasing the number of samples, as you noted. Both of the strange color patterns you  mention are really only issues for low sample densities, where undesirable behavior isn't really that surprising. 

This, and a few  of the the other graphics-related issues that are open right now, are the behavior of the graphics library that math3d uses [MathBox](https://github.com/unconed/mathbox). (I.e., this is the way that Mathbox handles color maps, and I believe it's done this way for performance reasons.)

**Please** don't think that means I'm uninterested in these issues. But, to be honest, it does mean they're less likely to be addressed. I am interested in moving to a a different graphics library, but that would be a long-term project, and other things (like user-login) are probably higher priorities. (For me, in my spare time, anyway).
